common/buildcraft/lib/fluids/TankUtils.java

```diff
@@ -4,19 +4,20 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
 
 public class TankUtils {
     public static void pushFluidAround(IBlockAccess world, BlockPos pos) {
         TileEntity tile = world.getTileEntity(pos);
-        IHasTank tileWithTank = (IHasTank) tile;
+        Tank tank = (Tank) tile.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
         for(EnumFacing side : EnumFacing.values()) {
             TileEntity tileToPush = world.getTileEntity(pos.offset(side));
-            if(tileToPush != null && tileToPush instanceof IHasTank) {
-                IHasTank tileToPushWithTank = (IHasTank) tileToPush;
-                int used = tileToPushWithTank.getTank().fill(tileWithTank.getTank().getFluid(), true);
+            if(tileToPush != null && tileToPush.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null)) {
+                Tank tankToPush = (Tank) tileToPush.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
+                int used = tankToPush.fill(tank.getFluid(), true);
 
                 if(used > 0) {
-                    tileWithTank.getTank().drain(used, true);
+                    tank.drain(used, true);
                 }
             }
```

Comment:
It would probably be better to change this so that you pass in the ````Tank```` variable rather than getting it via the capability (because at some point in the future the capability returned by pump needs to not accept any fluids externally.)